catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/ws/JsonRpc.java

```diff
@@ -97,83 +100,126 @@ public void onClose(Session session, int statusCode, String reason) {
 
   @Override
   public void onError(Session session, Throwable ex) {
-    // no action required on error
+    if (ex instanceof SecureWebSocketException) {
```

Comment:
A SecureWebSocketException sounds pretty general. Do we have any guarantees that a SecureWebSocketException always means the user is not logged in?

Reply:
Yeah I wasn't sure what to name it... I came up with this name before I fleshed out the implementation. The guarantee is just that this exception is only used by other classes in the same module 😉 Do you have a good suggestion? Maybe `WebSocketAuthenticationException`?

Reply:
Went with `WebSocketAuthenticationException`